portfolio/src/main/webapp/mapScript.js

```diff
@@ -0,0 +1,169 @@
+              stylers: [{color: '#9ca5b3'}]
+            },
+            {
+              featureType: 'road.highway',
+              elementType: 'geometry',
+              stylers: [{color: '#746855'}]
+            },
+            {
+              featureType: 'road.highway',
+              elementType: 'geometry.stroke',
+              stylers: [{color: '#1f2835'}]
+            },
+            {
+              featureType: 'road.highway',
+              elementType: 'labels.text.fill',
+              stylers: [{color: '#f3d19c'}]
+            },
+            {
+              featureType: 'transit',
+              elementType: 'geometry',
+              stylers: [{color: '#2f3948'}]
+            },
+            {
+              featureType: 'transit.station',
+              elementType: 'labels.text.fill',
+              stylers: [{color: '#d59563'}]
+            },
+            {
+              featureType: 'water',
+              elementType: 'geometry',
+              stylers: [{color: '#17263c'}]
+            },
+            {
+              featureType: 'water',
+              elementType: 'labels.text.fill',
+              stylers: [{color: '#515c6d'}]
+            },
+            {
+              featureType: 'water',
+              elementType: 'labels.text.stroke',
+              stylers: [{color: '#17263c'}]
+            }
+          ] 
+    });
+  map.addListener("click", event => {
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  });
+
+  fetchMarkers();
+}
+
+function fetchMarkers() {
+  fetch('/markers').then(response => response.json()).then(markers => {
+    markers.forEach(marker => {
+      createMarkerForDisplay(marker.lat, marker.lng, marker.content);
+    });
+  });
+}
+
+function postMarker(lat, lng, content) {
+  const params = new URLSearchParams();
+  params.append('lat', lat);
+  params.append('lng', lng);
+  params.append('content', content);
+  fetch('/markers', {method: 'POST', body: params});
+}
+
+function createMarkerForDisplay(lat, lng, content) {
+  const marker =
+      new google.maps.Marker({position: {lat: lat, lng: lng}, map: map});
+  const infoWindow = new google.maps.InfoWindow({content: content});
+  marker.addListener('click', () => {
+    infoWindow.open(map, marker);
+  });
+}
+
+function createMarkerForEdit(lat, lng, content) {
+  if (editMarker) {
+    editMarker.setMap(null);
+  } 
+  editMarker = 
+      new google.maps.Marker({position: {lat: lat, lng: lng}, map: map});
+  
+  const infoWindow =
+      new google.maps.InfoWindow({content: buildInfoWindowInput(lat, lng)});
+
+  google.maps.event.addListener(infoWindow, 'closeclick', () => {
+    editMarker.setMap(null);
+  });
+
+  infoWindow.open(map, editMarker);
+}
+
+function buildInfoWindowInput(lat, lng) {
+  const textBox = document.createElement('textarea');
+  const button = document.createElement('button');
+  button.appendChild(document.createTextNode('Submit'));
+
+  button.onclick = () => {
```

Comment:
It might help with readability if the onclick function isn't anonymous

Reply:
> It might help with readability if the onclick function isn't anonymous

This is probably more preference based, but imo two reasons why I prefer the anon function are: 1) The function code is right next to the event, so its easier to see exactly what happens when the onclick event is triggered, and 2) the anon function and surrounding variables share the same scope, so I don't have to pass any extra parameters into my eventhandler function.